### PR TITLE
Issue #7278: introduce a new message for the violation within import …

### PR DESCRIPTION
**Pull request by gmshake**
Address #7278

Regression report: https://gmshake.github.io/checkstyle_contribution/checkstyle-tester/diff_pr_7364/

### CODE REVIEW

**Comment by romani** [comment]
I see this as false positive - https://gmshake.github.io/checkstyle_contribution/checkstyle-tester/diff_pr_7364/lombok-ast/index.html#A2

```
1      package    packageDeclaration   ;
2   
3   import   java . util . * ;
4   import    static   java.util.Collections   .*;
5   import static java.lang.String . CASE_INSENSITIVE_ORDER;
6   
7   
8   
9   import java .
10  net.Socket    ;
```

static and non-static are separate groups. So original message is correct.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheckTest.java [hunk 1]
```diff
@@ -738,8 +739,8 @@ public void testNoPackage2() throws Exception {
         final String[] expected = {
             "7: " + getCheckMessage(MSG_LINE_SEPARATOR,
                 "com.sun.accessibility.internal.resources.*"),
-            "11: " + getCheckMessage(MSG_LINE_SEPARATOR, "java.util.Arrays"),
-            "19: " + getCheckMessage(MSG_LINE_SEPARATOR,
+            "11: " + getCheckMessage(MSG_SEPARATED_IN_GROUP, "java.util.Arrays"),
+            "19: " + getCheckMessage(MSG_SEPARATED_IN_GROUP,
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheckTest.java]: 1) please add test case from https://github.com/checkstyle/checkstyle/pull/7364#issuecomment-564741290 to our Input code base.

2) please add more test cases for different "customImportOrderRules", as right now we test it only on "STATIC###THIRD_PARTY_PACKAGE", please add few tests with different values for this property to make sure logic works fine for them too.

3) please add test input to `it` test part for google style coverage, as change affects behavior of of google style coverage. Inputs can be copied from `ut`.

Comment (gmshake) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheckTest.java]: Tests added.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+//non-compiled with javac: contains specially crafted set of imports for testing
+package com.puppycrawl.tools.checkstyle.checks.imports.customimportorder;
+
+import   java . util . * ;
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java]: What are the red blocks I see in github's difference view? It surrounds the `.`s on this line.

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java]: I think it is spaces, and it it error of highlighter

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java]:
```
✔ ~/java/github/checkstyle/checkstyle [gmshake/feature/issue_7278 L|✔] 
$ sed '4!d' src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java | xxd
00000000: 696d 706f 7274 2020 206a 6176 6120 2e20  import   java . 
00000010: 7574 696c 202e 202a 203b 0a              util . * ;.

✔ ~/java/github/checkstyle/checkstyle [gmshake/feature/issue_7278 L|✔] 
$ echo "java . u" | xxd
00000000: 6a61 7661 202e 2075 0a                   java . u.

✔ ~/java/github/checkstyle/checkstyle [gmshake/feature/issue_7278 L|✔] 
$ ascii -h
Usage: ascii [-dxohv] [-t] [char-alias...]
   -t = one-line output  -d = Decimal table  -o = octal table  -x = hex table
   -h = This help screen -v = version information
Prints all aliases of an ASCII character. Args may be chars, C \-escapes,
English names, ^-escapes, ASCII mnemonics, or numerics in decimal/octal/hex.

Dec Hex    Dec Hex    Dec Hex  Dec Hex  Dec Hex  Dec Hex   Dec Hex   Dec Hex  
  0 00 NUL  16 10 DLE  32 20    48 30 0  64 40 @  80 50 P   96 60 `  112 70 p
  1 01 SOH  17 11 DC1  33 21 !  49 31 1  65 41 A  81 51 Q   97 61 a  113 71 q
  2 02 STX  18 12 DC2  34 22 "  50 32 2  66 42 B  82 52 R   98 62 b  114 72 r
  3 03 ETX  19 13 DC3  35 23 #  51 33 3  67 43 C  83 53 S   99 63 c  115 73 s
  4 04 EOT  20 14 DC4  36 24 $  52 34 4  68 44 D  84 54 T  100 64 d  116 74 t
  5 05 ENQ  21 15 NAK  37 25 %  53 35 5  69 45 E  85 55 U  101 65 e  117 75 u
  6 06 ACK  22 16 SYN  38 26 &  54 36 6  70 46 F  86 56 V  102 66 f  118 76 v
  7 07 BEL  23 17 ETB  39 27 '  55 37 7  71 47 G  87 57 W  103 67 g  119 77 w
  8 08 BS   24 18 CAN  40 28 (  56 38 8  72 48 H  88 58 X  104 68 h  120 78 x
  9 09 HT   25 19 EM   41 29 )  57 39 9  73 49 I  89 59 Y  105 69 i  121 79 y
 10 0A LF   26 1A SUB  42 2A *  58 3A :  74 4A J  90 5A Z  106 6A j  122 7A z
 11 0B VT   27 1B ESC  43 2B +  59 3B ;  75 4B K  91 5B [  107 6B k  123 7B {
 12 0C FF   28 1C FS   44 2C ,  60 3C <  76 4C L  92 5C \  108 6C l  124 7C |
 13 0D CR   29 1D GS   45 2D -  61 3D =  77 4D M  93 5D ]  109 6D m  125 7D }
 14 0E SO   30 1E RS   46 2E .  62 3E >  78 4E N  94 5E ^  110 6E n  126 7E ~
 15 0F SI   31 1F US   47 2F /  63 3F ?  79 4F O  95 5F _  111 6F o  127 7F DEL

```

Comment (gmshake) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java]: I've re-checked with `hexdump`, they're spaces.

```
zlei$ hexdump -C src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java
```
```
00000000  2f 2f 6e 6f 6e 2d 63 6f  6d 70 69 6c 65 64 20 77  |//non-compiled w|
00000010  69 74 68 20 6a 61 76 61  63 3a 20 63 6f 6e 74 61  |ith javac: conta|
00000020  69 6e 73 20 73 70 65 63  69 61 6c 6c 79 20 63 72  |ins specially cr|
00000030  61 66 74 65 64 20 73 65  74 20 6f 66 20 69 6d 70  |afted set of imp|
00000040  6f 72 74 73 20 66 6f 72  20 74 65 73 74 69 6e 67  |orts for testing|
00000050  0a 70 61 63 6b 61 67 65  20 63 6f 6d 2e 70 75 70  |.package com.pup|
00000060  70 79 63 72 61 77 6c 2e  74 6f 6f 6c 73 2e 63 68  |pycrawl.tools.ch|
00000070  65 63 6b 73 74 79 6c 65  2e 63 68 65 63 6b 73 2e  |eckstyle.checks.|
00000080  69 6d 70 6f 72 74 73 2e  63 75 73 74 6f 6d 69 6d  |imports.customim|
00000090  70 6f 72 74 6f 72 64 65  72 3b 0a 0a 69 6d 70 6f  |portorder;..impo|
000000a0  72 74 20 20 20 6a 61 76  61 20 2e 20 75 74 69 6c  |rt   java . util|
000000b0  20 2e 20 2a 20 3b 0a 69  6d 70 6f 72 74 20 20 20  | . * ;.import   |
000000c0  20 73 74 61 74 69 63 20  20 20 6a 61 76 61 2e 75  | static   java.u|
000000d0  74 69 6c 2e 43 6f 6c 6c  65 63 74 69 6f 6e 73 20  |til.Collections |
000000e0  20 20 2e 2a 3b 0a 69 6d  70 6f 72 74 20 73 74 61  |  .*;.import sta|
000000f0  74 69 63 20 6a 61 76 61  2e 6c 61 6e 67 2e 53 74  |tic java.lang.St|
00000100  72 69 6e 67 20 2e 20 43  41 53 45 5f 49 4e 53 45  |ring . CASE_INSE|
00000110  4e 53 49 54 49 56 45 5f  4f 52 44 45 52 3b 0a 0a  |NSITIVE_ORDER;..|
00000120  0a 69 6d 70 6f 72 74 20  6a 61 76 61 20 2e 0a 09  |.import java ...|
00000130  09 6e 65 74 2e 53 6f 63  6b 65 74 20 20 20 20 3b  |.net.Socket    ;|
00000140  0a 0a 70 75 62 6c 69 63  20 63 6c 61 73 73 20 49  |..public class I|
00000150  6e 70 75 74 43 75 73 74  6f 6d 49 6d 70 6f 72 74  |nputCustomImport|
00000160  4f 72 64 65 72 56 69 6f  6c 61 74 69 6f 6e 73 53  |OrderViolationsS|
00000170  61 6d 65 4c 69 6e 65 20  7b 7d 0a                 |ameLine {}.|
0000017b
```

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/imports/customimportorder/InputCustomImportOrderViolationsSameLine.java]: Thanks for confirming. Github made it look like it was unprintable characters.

**Pull request merged**